// @shopify/app/dist/cli/services/conf.js
import { Conf } from '@shopify/cli-kit/node/conf';
import { outputDebug, outputContent, outputToken } from '@shopify/cli-kit/node/output';
let _instance;
function appConf() {
    if (!_instance) {
        _instance = new Conf({ projectName: 'shopify-cli-app' });
    }
    return _instance;
}
export function getAppInfo(directory, config = appConf()) {
    outputDebug(outputContent `Reading cached app information for directory ${outputToken.path(directory)}...`);
    return config.get(directory);
}
export function clearAppInfo(directory, config = appConf()) {
    outputDebug(outputContent `Clearing app information for directory ${outputToken.path(directory)}...`);
    config.delete(directory);
}
export function clearAllAppInfo(config = appConf()) {
    outputDebug(outputContent `Clearing all app information...`);
    config.clear();
}
export function setAppInfo(options, config = appConf()) {
    outputDebug(outputContent `Storing app information for directory ${outputToken.path(options.directory)}:${outputToken.json(options)}`);
    const savedApp = config.get(options.directory);
    if (savedApp) {
        config.set(options.directory, {
            directory: options.directory,
            appId: options.appId ?? savedApp.appId,
            title: options.title ?? savedApp.title,
            storeFqdn: options.storeFqdn ?? savedApp.storeFqdn,
            orgId: options.orgId ?? savedApp.orgId,
            updateURLs: options.updateURLs ?? savedApp.updateURLs,
            tunnelPlugin: options.tunnelPlugin ?? savedApp.tunnelPlugin,
        });
    }
    else {
        config.set(options.directory, options);
    }
}
